xdsl/ir/core.py

```diff
@@ -309,12 +300,27 @@ class MyEnumAttribute(EnumAttribute[MyEnum], SpacedOpaqueSyntaxAttribute):
     """
 
     enum_type: ClassVar[type[StrEnum]]
+    enum_to_string: dict[StrEnum, str]
```

Comment:
I'm not sure of the value of this performance optimisation, can't we wait until we see enum printing to be slow before speeding it up? Isn't printing more of a debug thing anyway?